src/main/java/Duke.java

```diff
@@ -1,10 +1,320 @@
+// import libraries here
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.nio.file.*;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.Set;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class Duke {
+    private static ArrayList<String> descriptions = new ArrayList<>();
+    private static ArrayList<Boolean> dones = new ArrayList<>();
+    private static ArrayList<String> types = new ArrayList<>();
+    private static ArrayList<String> dates = new ArrayList<>();
```

Comment:
Suggest to distribute the logic here to the class definitions. This approach conflicts with OOP.